tests/baselines/reference/privateNameNestedClassNameConflict.js

```diff
@@ -0,0 +1,28 @@
+//// [privateNameNestedClassNameConflict.ts]
+class A {
```

Comment:
1. probably should be named `privateNameNestedClassNameShadow.ts` if there's no error.
2. Why does the inner weakmap need to have a unique name?
2. What is the emit for

```ts
class A {
  #foo = 1
}
class B {
  #foo = 1
}
```

Do they both share a weakmap `_foo`? Or does B.#foo get a unique weakmap name?

Reply:
privateNamesAndFields answers (3): They do not share a weakmap; B gets `_foo_1`.